src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -322,19 +322,12 @@ private static byte[] DecodeSignature(CborReader reader)
         private static List<CoseSignature> DecodeCoseSignaturesArray(CborReader reader, byte[] bodyProtected)
         {
             int? signaturesLength = reader.ReadStartArray();
+            List<CoseSignature> signatures = new List<CoseSignature>(signaturesLength ?? 1);
```

Comment:
Is `1` the best fallback value here?  Given that it's a multi, at least one is expected, and two or three doesn't seem unheard of.

I think I'd expect one of

* 0: Let List decide its own default (4)
* 4: List's default, bigger than the number we expect.
* 10: Much bigger than we expect.
* Doing some pre-counting to just get the right answer.

Reply:
> Is 1 the best fallback value here?

Yes, this is reading a [+COSE_Signature] i.e: the array of signatures, which is required to have at least one. If we have more, the list will grow, if we have less (zero), we will throw.

Reply:
Sure, at least one item is expected, but why are we putting the List into its worst possible state of growth?

If you make a List(1), then it has 3 entries, it goes

* new List(1)
  * `storage = new T[1];`
* Add 1
* Add 2
  * Too small! Double in size.  `new T[2]`
* Add 3
  * Too small! Double in size.  `new T[4]`

`new List<T>(0)` says "choose your own default size" (which is 4); and any number other than 1 skips the 1->2 growth phase.

I agree that 1 is the most likely value for Count; but I think we should use literally any other number for initial capacity.

Reply:
> new List<T>(0) says "choose your own default size" (which is 4); and any number other than 1 skips the 1->2 growth phase.

Ah, I didn't know that was how `List`'s growth worked. In that case, we're better off letting the List decide its own default.